Firebase/Database/Core/FServerValues.m

```diff
@@ -69,22 +145,28 @@ + (id)resolveDeferredValueTree:(FSparseSnapshotTree *)tree
                              rememberData:
                                  [FServerValues
                                      resolveDeferredValueSnapshot:node
-                                                 withServerValues:serverValues]
+                                                     withExisting:
+                                                         [existing
+                                                             getChild:path]
```

Comment:
This `[existing getChild:path]` is the line that was missing from the JS SDK. We could add the rejected performance optimizations with a lazy-evaluation of data, but then we should check it into JS too.